### PR TITLE
chore: upgrade btcd (and tss deps)

### LINKED ISSUE #2728: Upgrade `btcsuite` project-wise
We need to upgrade `btcsuite` in the node and in `go-tss` to enable Taproot scripts. Currently, we have several different `btcd` package versions, which creates a headache in BTC dependencies in the project. We need to make it simpler.

Blocks https://github.com/zeta-chain/node/pull/2727

### PR DESCRIPTION
**Pull request by gartnera**
Upgrade github.com/btcsuite/btcd. Use [tss-lib fork](https://github.com/zeta-chain/tss-lib/tree/threshold-dep-updates) which upgrades [btcd](https://github.com/zeta-chain/tss-lib/commit/2e6b438bd9019c5918a6b180ed0cefea534d1da1).

TODO:

- [x] restore HSM code [done after upgrading keystone](https://github.com/zeta-chain/keystone/commit/3874f358c138c872632528fe7f6d36fb5c0bc861)
- [x] merge https://github.com/zeta-chain/go-tss/pull/23

Deferred items:
- simplify key conversion logic
- improve BIP 341/342 compliance by removing `txscript.NewCannedPrevOutputFetcher([]byte{}, 0)`

Should probably merge https://github.com/zeta-chain/node/pull/2767 rapidly after this to get on the latest tss-lib. Update: this work was merged into this change.

Closes #2728

## Summary by CodeRabbit

- **New Features**
	- Updated key generation process with a new import for the `keygen` package, enhancing functionality.
	- Simplified the transaction signing process by modifying the `SignTx` method. 

- **Bug Fixes**
	- Improved error messaging in test functions for better clarity.

- **Chores**
	- Updated import paths for various packages to ensure compatibility and maintainability.
	- Updated version numbers of dependencies to their latest releases for improved performance and security.

### CODE REVIEW

**Review by lumtis — APPROVED** [review]
There are TODOs not checked in PR description, is it beyond the scope of this PR? If so can we create issues for those if there are none?

File: cmd/zetaclientd/hsm.go [hunk 1]
```diff
@@ -1,105 +0,0 @@
-package main
```
[the remark this answers is not in the document]
Comment (gartnera) [reply cmd/zetaclientd/hsm.go:1]: Yep I set it as ready for review as I was looking for initial reviews.

File: go.mod [hunk 2]
```diff
@@ -366,4 +362,6 @@ replace github.com/evmos/ethermint => github.com/zeta-chain/ethermint v0.0.0-202
 
 replace github.com/libp2p/go-libp2p => github.com/zeta-chain/go-libp2p v0.0.0-20240710192637-567fbaacc2b4
 
-replace gitlab.com/thorchain/tss/go-tss => github.com/zeta-chain/go-tss v0.0.0-20240729195411-9f5ae8189449
+replace gitlab.com/thorchain/tss/go-tss => github.com/zeta-chain/go-tss v0.0.0-20240819223347-fce649aca224
+
+replace github.com/bnb-chain/tss-lib => github.com/zeta-chain/tss-lib v0.0.0-20240819214903-774ccaf59fca
```
[the remark this answers is not in the document]
Comment (gartnera) [reply go.mod:367]: I've updated the description with the details, but it's part of https://github.com/zeta-chain/go-tss/pull/23

File: pkg/chains/address_test.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ func TestAddress(t *testing.T) {
 func TestDecodeBtcAddress(t *testing.T) {
 	t.Run("invalid string", func(t *testing.T) {
 		_, err := DecodeBtcAddress("�U�ڷ���i߭����꿚�l", BitcoinTestnet.ChainId)
-		require.ErrorContains(t, err, "runtime error: index out of range")
```
Comment (brewmaster012) [inline pkg/chains/address_test.go:48]: interesting, seems they fixed something here...

**Review by brewmaster012 — APPROVED** [review]
LGTM

**Pull request merged**